Looks like it's unhappy on windows; if you have wine, you can reproduce it without windows, e.g.
```
$ GOOS=windows tinygo test -c os
$ wine os.test -test.v -test.run TestFd
=== RUN   TestFd
--- FAIL: TestFd (0.00s)
    Failed to read 5 bytes from file descriptor: readat as-fd: operation not implemented
    File descriptor contents not equal to file contents.
FAIL
```
This is not surprising, as fds are very different on windows, and upstream's "pollfd" layer that hides the difference (I think) has not been ported yet.

So just skip that test on windows, as TestMkdirTempBadDir() does.

--- a/src/os/file.go
+++ b/src/os/file.go
@@ -179,9 +179,19 @@ func (f *File) SyscallConn() (syscall.RawConn, error) {
 	return nil, ErrNotImplemented
 }
 
+// fd is an internal interface that is used to try a type assertion in order to
+// call the Fd() method of the underlying file handle if it is implemented.
+type fd interface {
+	Fd() uintptr
+}
+
 // Fd returns the file handle referencing the open file.
 func (f *File) Fd() uintptr {
-	panic("unimplemented: os.file.Fd()")
+	handle, ok := f.handle.(fd)
+	if ok {
+		return handle.Fd()
+	}
+	return 0
 }
 
 // Truncate is a stub, not yet implemented

--- a/src/os/file_anyos.go
+++ b/src/os/file_anyos.go
@@ -118,6 +118,10 @@ func (f unixFileHandle) Close() error {
 	return handleSyscallError(syscall.Close(syscallFd(f)))
 }
 
+func (f unixFileHandle) Fd() uintptr {
+	return uintptr(f)
+}
+
 // Chmod changes the mode of the named file to mode.
 // If the file is a symbolic link, it changes the mode of the link's target.
 // If there is an error, it will be of type *PathError.

--- a/src/os/file_anyos_test.go
+++ b/src/os/file_anyos_test.go
@@ -68,3 +68,21 @@ func TestChdir(t *testing.T) {
 		t.Errorf("Remove %s: %s", dir, err)
 	}
 }
+
+func TestStandardFd(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Log("TODO: TestFd fails on Windows, skipping")
+		return
+	}
+	if fd := Stdin.Fd(); fd != 0 {
+		t.Errorf("Stdin.Fd() = %d, want 0", fd)
+	}
+
+	if fd := Stdout.Fd(); fd != 1 {
+		t.Errorf("Stdout.Fd() = %d, want 1", fd)
+	}
+
+	if fd := Stderr.Fd(); fd != 2 {
+		t.Errorf("Stderr.Fd() = %d, want 2", fd)
+	}
+}

--- a/src/os/file_anyos_test.go
+++ b/src/os/file_anyos_test.go
@@ -4,7 +4,9 @@
 package os_test
 
 import (
+	"io"
 	. "os"
+	"runtime"
 	"testing"
 )
 
@@ -84,5 +86,31 @@ func TestStandardFd(t *testing.T) {
 
 	if fd := Stderr.Fd(); fd != 2 {
 		t.Errorf("Stderr.Fd() = %d, want 2", fd)
+	}
+}
+
+func TestFd(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Log("TODO: TestFd fails on Windows, skipping")
+		return
+	}
+	f := newFile("TestFd.txt", t)
+	defer Remove(f.Name())
+	defer f.Close()
+
+	const data = "hello, world\n"
+	io.WriteString(f, data)
+
+	fd := NewFile(f.Fd(), "as-fd")
+	defer fd.Close()
+
+	b := make([]byte, 5)
+	n, err := fd.ReadAt(b, 0)
+	if n != 5 && err != nil {
+		t.Errorf("Failed to read 5 bytes from file descriptor: %v", err)
+	}
+
+	if string(b) != data[:5] {
+		t.Errorf("File descriptor contents not equal to file contents.")
 	}
 }

--- a/src/os/file_other.go
+++ b/src/os/file_other.go
@@ -71,6 +71,10 @@ func (f stdioFileHandle) Seek(offset int64, whence int) (int64, error) {
 	return -1, ErrUnsupported
 }
 
+func (f stdioFileHandle) Fd() uintptr {
+	return uintptr(f)
+}
+
 //go:linkname putchar runtime.putchar
 func putchar(c byte)
 

